ceph/mds.go

```diff
@@ -0,0 +1,173 @@
+package ceph
+
+import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	cephCmd                      = "/usr/bin/ceph"
+	mdsBackgroundCollectInterval = 5 * time.Minute
+)
+
+const (
+	MDSModeDisabled   = 0
+	MDSModeForeground = 1
+	MDSModeBackground = 2
+)
+
+type mdsStat struct {
+	FSMap struct {
+		Filesystems []struct {
+			MDSMap struct {
+				FSName string `json:"fs_name"`
+				Info   map[string]struct {
+					GID   uint   `json:"gid"`
+					Name  string `json:"name"`
+					Rank  int    `json:"rank"`
+					State string `json:"state"`
```

Comment:
per-MDS fields to add in future PR:
- export_targets
- incarnation (epoch where this MDS went live/non-standby)

Reply:
100%!